vscripts/gamemodes/custom_ctf/cl_gamemode_custom_ctf.nut

```diff
@@ -642,14 +660,180 @@ void function waitrespawn(entity player)
+	backgroundModelGeo.kv.disableshadows = 1
+	backgroundModelGeo.kv.fadedist = -1
+	backgroundModelGeo.MakeSafeForUIScriptHack()
+
+	backgroundModelSmoke = CreateClientSidePropDynamic( targetBackground.GetOrigin() - <0, 0, 24>, targetBackground.GetAngles(), $"mdl/levels_terrain/mp_lobby/mp_character_select_smoke.rmdl" )
+	backgroundModelSmoke.kv.solid = 0
+	backgroundModelSmoke.kv.disableshadows = 1
+	backgroundModelSmoke.kv.fadedist = -1
+	backgroundModelSmoke.MakeSafeForUIScriptHack()
+
+    backgroundModelGeo.SetSkin( 3 )
+    backgroundModelSmoke.SetSkin( 3 )
+
+    vector cameraOrigin = targetCamera.GetOrigin()
+    votecamera = CreateClientSidePointCamera( cameraOrigin, targetCamera.GetAngles(), 35.5 )
+
+    GetLocalClientPlayer().SetMenuCameraEntity( votecamera )
+	votecamera.SetTargetFOV( 35.5, true, EASING_CUBIC_INOUT, 0.25 )
+
+    RunUIScript( "OpenCTFVoteMenu" )
+
+    ScreenFade(GetLocalClientPlayer(), 0, 0, 0, 255, 0.3, 0.0, FFADE_IN | FFADE_PURGE)
+}
+
+void function ServerCallback_CTF_CloseCTFVoteMenu()
+{
+    thread DestroyVotingUI()
+}
+
+void function DestroyVotingUI()
+{
+    FadeOutSoundOnEntity( GetLocalClientPlayer(), "Music_CharacterSelect_Wattson", 0.2 )
+    //EmitSoundOnEntity( GetLocalClientPlayer(), "UI_InGame_FD_UnReadyUp_1p" ) //UI_InGame_FD_MetaUpgradeAnnouncement
+    ScreenFade(GetLocalClientPlayer(), 0, 0, 0, 255, 0.4, 1, FFADE_OUT | FFADE_PURGE)
+    wait 1;
+
+    if ( IsValid( backgroundModelSmoke ) )
+		backgroundModelSmoke.Destroy()
+	if ( IsValid( backgroundModelGeo ) )
+		backgroundModelGeo.Destroy()
+    if ( IsValid( votecamera ) )
+		votecamera.Destroy()
+
+    GetLocalClientPlayer().ClearMenuCameraEntity()
+
+    RunUIScript( "CloseCTFVoteMenu" )
+
+    ScreenFade(GetLocalClientPlayer(), 0, 0, 0, 255, 0.3, 0.5, FFADE_IN | FFADE_PURGE)
+}
+
+void function UpdateUIVoteTimer()
+{
+    int time = 15
+    while(time > -1)
+    {
+        RunUIScript( "UpdateVoteTimer", time)
+
+        if (time <= 5 && time != 0)
+            EmitSoundOnEntity( GetLocalClientPlayer(), "ui_ingame_markedfordeath_countdowntomarked" )
+
+        if (time == 0)
+            EmitSoundOnEntity( GetLocalClientPlayer(), "ui_ingame_markedfordeath_countdowntoyouaremarked" )
+
+        time--
+
+        wait 1
+    }
+}
+
+void function UpdateUIVotingLocationDone(int mapid)
+{
+    EmitSoundOnEntity( GetLocalClientPlayer(), "UI_PostGame_Level_Up_Pilot" )
+    RunUIScript( "UpdateVotedLocation", file.locationSettings[mapid].name)
+}
+
+void function UpdateUIVotingLocationTied(int mapid, int done)
+{
+    if (done == 0)
+        EmitSoundOnEntity( GetLocalClientPlayer(), "HUD_match_start_timer_tick_1P" )
+
+    if (done == 1)
+        EmitSoundOnEntity( GetLocalClientPlayer(), "UI_PostGame_CoinMove" )
+
+    if (mapid == 254)
+        RunUIScript( "UpdateVotedLocationTied", "")
+    else
+        RunUIScript( "UpdateVotedLocationTied", file.locationSettings[mapid].name)
+}
+
+void function VoteForMap(int mapid)
+{
+    if(hasvoted)
```

Comment:
Maybe don't rely on the client to specify if we already voted.

Reply:
Handle it server side.

Reply:
it is checked server side as well